ml-agents/mlagents/trainers/sac/optimizer_torch.py

```diff
@@ -58,21 +59,29 @@ def forward(
             actions: Optional[torch.Tensor] = None,
             memories: Optional[torch.Tensor] = None,
             sequence_length: int = 1,
+            q1_grad: bool = True,
+            q2_grad: bool = True,
         ) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
```

Comment:
This forward method is getting rather complex, I would add a docstring to it for posterity.